src/main/kotlin/org/rust/lang/doc/lexer/RustDocHighlightingLexer.flex

```diff
@@ -0,0 +1,179 @@
+package org.rust.lang.doc.lexer;
+
+import com.intellij.lexer.*;
+import com.intellij.psi.tree.IElementType;
+
+import static org.rust.lang.doc.psi.RustDocElementTypes.*;
+import static com.intellij.psi.TokenType.*;
+
+%%
+
+%{
+  private int MAIN_STATE = YYINITIAL;
+
+  public _RustDocHighlightingLexer(boolean isBlock) {
+    this((java.io.Reader)null);
+    MAIN_STATE = isBlock ? IN_BLOCK : IN_EOL;
+  }
+
+  // FIXME(jajakobyly): I think it is possible this will break, though I couldn't make it do so :-)
+  private char CODE_FENCE_DELIM = '\0';
+
+  private boolean isLastToken() {
+    return zzMarkedPos == zzEndRead;
+  }
+
+  private void docHeadingTrimRight() {
+    int i = yylength() - 1;
+    char ch = yycharat(i);
+
+    // Trim trailing *****/ if we are at the end of block doc comment
+    if (i >= 1 && MAIN_STATE == IN_BLOCK && isLastToken() && ch == '/' && yycharat(i - 1) == '*') {
+      i -= 2;
```

Comment:
Hmm, actually I can't answer this question :sweat_smile: Will have to revise this condition